bids/layout/index.py

```diff
@@ -164,10 +165,37 @@ def index_files(self):
         """Index all files in the BIDS dataset. """
         self._index_dir(self.root, self.config)
 
-    def index_metadata(self):
+    def index_metadata(self, **filters):
         """Index metadata for all files in the BIDS dataset. """
+        if filters:
+            default_ext = ['[.]+']
+            # ensure we are returning objects
+            filters['return_type'] = 'object'
+            # until 0.11.0, user can specify extension or extensions
+            if filters.get('extension'):
+                ext_key = 'extension'
+            elif filters.get('extensions'):
+                ext_key = 'extensions'
+            else:
+                ext_key = 'extension'
+                msg = (
+                    "You should explicitly set the extension argument. "
```

Comment:
Why does the `extension` key need to be set to that regex? It seems in `BIDSLayout.__init__` setting it to None works

Reply:
From my use of `pybids` setting a key to `None` means do not match any files that contain this key, so my interpretation of `extension=None` would be to match files without an extension. When I set extensions to None and ran the tests, when this test [would fail](https://github.com/bids-standard/pybids/pull/560/files#diff-0f4955beecc5dc480769ecddccfdab39R30) since none of the nifti files were indexed.

Reply:
Ah, yeah I guess there's a difference between setting a key to `None` versus not setting a key at all. So what happens if you don't set it at all?

Reply:
Just saying because if `filters` is None, then `kwargs` does not have extensions, yet all meta-data is indexed, correct?

Reply:
That's a good idea, looks like it indexes the correct information, allowing me to remove the weird logic around what should be specified in the `extension` key.

Reply:
That would be nice because then we could cut this whole `if` clause out.